pkgs/stdenv/generic/setup.sh

```diff
@@ -995,10 +995,18 @@ _defaultUnpack() {
 
     if [ -d "$fn" ]; then
 
+        destination="$(stripHash "$fn")"
```

Comment:
I suggest declaring `destination` as `local` at the beginning of the function to make it lexically scoped.

Reply:
Done.